src/wc/wc.rs

```diff
@@ -133,6 +160,31 @@ fn is_word_seperator(byte: u8) -> bool {
     byte == SPACE || byte == TAB || byte == CR || byte == SYN || byte == FF
 }
 
+#[inline]
+#[cfg(any(target_os = "linux", target_os = "android"))]
+fn count_bytes_using_splice(fd: RawFd) -> nix::Result<usize> {
+    const BUF_SIZE: usize = 16 * 1024;
+
+    let null_file = OpenOptions::new()
+        .write(true)
+        .open("/dev/null")
```

Comment:
If you do what I mentioned below, this could be added as a zero-copy `Sink` or something.